src/controllers/usersController.js

```diff
@@ -19,10 +22,48 @@ exports.registeration = (req, res) => {
     res.json(err);
   });
 };
-exports.login = (req, res) => {
-// return token
-  res.json('login in progress');
-};
+
+exports.login = [
+  [
+    // username must be an email
+    check('email').isEmail().withMessage('Invalid email addres'),
+    // password must be at least 5 chars long
+    check('password').isLength({ min: 5 }).withMessage('Password must be at least 5 chars long')
+  ], async (req, res) => {
+    // Finds the validation errors in this request and wraps them in an object with handy functions
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ status: false, errors: errors.array() });
+    }
+    const user = await User.findOne({
```

Comment:
i cant see the findOne method as a change in the users class

Reply:
I don't get this...